plcc/src/proteinstructure/Molecule.java

```diff
@@ -482,12 +494,118 @@ private Integer centroidDistTo(Molecule m) {
+     * which all Atoms are located. If this distance + 2 does not overlap with the one of the compared Molecule, they cannot be in contact.
+     */
+
+//    public <T extends Molecule> Boolean contactPossibleWithResidue(T m) {
+//        // ugly code: instead of instanceof some generic methods should be used
+//        if (m instanceof Residue) {
+//            Residue r;
+//            r = (Residue) m;
+//            
+//            Integer dist = Integer.MAX_VALUE;
+//            try {
+//                dist = this.getCenterAtom().distToAtom(r.getCenterAtom());
+//            }
+//            catch(Exception e) {
+//                if( ! Settings.getBoolean("plcc_B_no_parse_warn")) {
+//                    DP.getInstance().w("Could not determine distance between DSSP residues " + this.getDsspNum() + " and " + r.getDsspNum() + ", assuming out of contact distance.");
+//                }
+//                return(false);
+//            }
+//            Integer atomRadius;
+//            if(this.isLigand() || r.isLigand()) {
+//                atomRadius = Settings.getInteger("plcc_I_lig_atom_radius");
+//            }
+//            else if (this.isAA() || r.isAA()) {
+//                atomRadius = Settings.getInteger("plcc_I_aa_atom_radius");
+//            }
+//            else {
+//                atomRadius = Settings.getInteger("plcc_I_rna_atom_radius");
+//            }
+//
+//            Integer justToBeSure = 4;   // Setting this to 0 shouldn't change the number of contacts found (but all harm it could do is to increase the runtime a tiny bit). Verified: has no influence. Should be removed in future release.
+//            Integer maxDistForContact = this.getCenterSphereRadius() + r.getCenterSphereRadius() +  (atomRadius * 2) + justToBeSure;
+//
+//            //System.out.println("    Center sphere radius for PDB residue " + this.getPdbResNum() + " = " + this.getCenterSphereRadius() + ", for " + r.getPdbResNum() + " = " + r.getCenterSphereRadius() + ", atom radius is " + atomRadius + ".");
+//            //System.out.println("    DSSP Res distance " + this.getDsspResNum() + "/" + r.getDsspResNum() + " is " + dist + " (no contacts possible above distance " + maxDistForContact + ").");
+//
+//            if(dist > (maxDistForContact)) {
+//                return(false);
+//            }
+//            else {
+//                return(true);
+//            }
+//        }
+//        else {
+//            DP.getInstance().w("Tried to get distance between Molecules. Due to ugly code this is not possible atm.");
+//        }
+//
+//        // if one of them is Molecule or RNA
+//        return false;
+//    }
 
     // ugly abstract methods
-    abstract public Integer getType();
     abstract public Boolean isAA();
     abstract public Boolean isLigand();
+    
+
+    /**
+     * Returns the type-specific atom radius of the component (type = Residue, Ligand, Other, RNA).
```

Comment:
I struggle with the "type = Residue ..." part. One could think because of the equal sign that type is a string of these values. I suggest writing it either as "type stands for..." or repeat the code.